### PR TITLE
Fix pipe docs

### LINKED ISSUE #27054: DOC: Fix pipe example in the Essential basic functionality page
In our continuous integration we validate that all the examples in the documentation follow PEP-8, and are valid code that can run.

Currently, we're skipping the essential basic functionality page, because the examples showing the `pipe` functionality can't be run, because the variables it uses are not defined. The example can be seen here:
https://github.com/pandas-dev/pandas/blame/master/doc/source/getting_started/basics.rst#L765

What we should do is to replace this example, by a real-world example, where the data is previously created. This way users will be able to run the code, and also will understand better a real-world case where `.pipe()` can be used.

A great resource for understanding `pipe` and to get inspiration for an example is this blog post: https://tomaugspurger.github.io/method-chaining

What needs to be done here is:
- Add a **minimalistic** DataFrame with somehow real-world data.
- Create couple of **one-liner** functions that can be used with pipe and the dataset created, in a meaningful way.
- Make sure that the example is not overcomplicated, and that the data and the functions are so simple that the reader can focus all their attention to understanding `pipe` and method chaining.
- Start validating the code in the file (i.e. remove this line: https://github.com/pandas-dev/pandas/blob/master/setup.cfg#L49)

### PR DESCRIPTION
**Pull request by alimcmaster1**
- [x] closes #27054

See the image below which highlights what i've done here

image

`flake8-rst doc/source --filename=/getting_started/basics.rst` runs clean

cc. @datapythonista to review

### CODE REVIEW

**Comment by alimcmaster1** [comment]
Unsure why the code coverage has dramatically reduced..

File: doc/source/getting_started/basics.rst [hunk 1]
```diff
@@ -753,28 +753,51 @@ on an entire ``DataFrame`` or ``Series``, row- or column-wise, or elementwise.
...
+
+.. ipython:: python
 
-   # f, g, and h are functions taking and returning ``DataFrames``
-   >>> f(g(h(df), arg1=1), arg2=2, arg3=3)
+    def extract_city_name(df):
+        """
+        Chicago, IL -> Chicago for city_name column
+        """
+        df['city_name'] = df['city_and_code'].apply(lambda x: x.split(', ')[0])
```
Comment (WillAyd) [inline doc/source/getting_started/basics.rst]:
```suggestion
        df['city_name'] = df['city_and_code'].str.split(",").str.get(0)
```

File: doc/source/getting_started/basics.rst [hunk 2]
```diff
@@ -753,28 +753,51 @@ on an entire ``DataFrame`` or ``Series``, row- or column-wise, or elementwise.
...
+        return df
 
-with the equivalent
+    def add_country_name(df, country_name=None):
+        """
+        Chicago -> Chicago-US for city_name column
+        """
+        col = 'city_name'
+        df['city_and_country'] = df[col].apply(lambda x: f'{x}-{country_name}')
```
Comment (WillAyd) [inline doc/source/getting_started/basics.rst]:
```suggestion
        df['city_and_country'] = df[col] + "-USA"
```

**Review by WillAyd — CHANGES_REQUESTED** [review]
Want to avoid using lambdas where possible

**Comment by alimcmaster1** [comment]
Thanks for the comments @WillAyd totally agree - updated

**Review by WillAyd — APPROVED** [review]
lgtm @datapythonista care to take a look here?

**Comment by WillAyd** [comment]
Thanks @alimcmaster1

**Comment by alimcmaster1** [comment]
Thanks @WillAyd

**Comment by datapythonista** [comment]
Sorry I couldn't check this before. Looks great, good job @alimcmaster1. Happy to see this finally improved.

**Pull request merged**